qiskit/aqua/algorithms/factorizers/shor.py

```diff
@@ -299,85 +302,77 @@ def _get_factors(self, output_desired, t_upper):
-                return False
-
-            # If the value is not to big (infinity),
-            # then get the right values and do the proper gcd()
-            putting_plus = int(exponential + 1)
-            putting_minus = int(exponential - 1)
-            one_factor = math.gcd(putting_plus, self._N)
-            other_factor = math.gcd(putting_minus, self._N)
-
-            # Check if the factors found are trivial factors or are the desired factors
-            if one_factor == 1 or one_factor == self._N or \
-                    other_factor == 1 or other_factor == self._N:
-                logger.debug('Found just trivial factors, not good enough.')
-                # Check if the number has already been found,
-                # use i-1 because i was already incremented
-                if t[i - 1] == 0:
-                    self._ret['results'][output_desired] = \
-                        'the continued fractions found exactly x_final/(2^(2n)).'
-                    return False
-                if i >= self._N:
-                    self._ret['results'][output_desired] = \
-                        'unable to find factors after too many attempts.'
-                    return False
+            if exponential > 1000000000:
+                fail_reason = 'denominator of continued fraction is too big.'
             else:
-                logger.debug('The factors of %s are %s and %s.', self._N, one_factor, other_factor)
-                logger.debug('Found the desired factors.')
-                self._ret['results'][output_desired] = (one_factor, other_factor)
-                factors = sorted((one_factor, other_factor))
-                if factors not in self._ret['factors']:
-                    self._ret['factors'].append(factors)
-                return True
-
-    def _run(self):
+                # The value is not too big,
+                # get the right values and do the proper gcd()
+                putting_plus = int(exponential + 1)
+                putting_minus = int(exponential - 1)
+                one_factor = math.gcd(putting_plus, self._N)
+                other_factor = math.gcd(putting_minus, self._N)
+
+                # Check if the factors found are trivial factors or are the desired factors
+                if any([factor in {1, self._N} for factor in (one_factor, other_factor)]):
+                    logger.debug('Found just trivial factors, not good enough.')
+                    # Check if the number has already been found,
+                    # (use i - 1 because i was already incremented)
+                    if t[i - 1] == 0:
+                        fail_reason = 'the continued fractions found exactly x_final/(2^(2n)).'
+                else:
+                    # Successfully factorized N
+                    return sorted((one_factor, other_factor))
+
+        # Search for factors failed, write the reason for failure to the debug logs
+        logger.debug(
```

Comment:
We only get to this point if the loop ends, which is either too many attempts or a failure. Rather than logging an ambiguous message if we get here i.e fail reason or it took too many... can we instead detect if fail reason is None and if so set fail_reason to 'it took too many....' so a single reason can be logged.

I also note that the loop here continues while i <= N (i.e. breaks when i > N) but in the former code it returned (ie broke) when i >= N  with too many attempts.

Reply:
Maybe I'm missing the point of the first issue there, however the two options
```python
logged_reason = fail reason or "too many attempts"
```
or
```python
if fail_reason is None:
    logged_reason = "too many attempts"
```
both equate to the same thing.  When would the ambiguous message be logged?

As for the second issue, yes the limit `i = N` is evaluated in this new code.  The former code also evaluated the factors when `i = N`, halting when the denominator was odd or trivial factors were found.  Those are equivalent to the exit points of this new loop, so I think this change is fine, but if there's a fault in my logic then I'm happy to change it

Reply:
Ahh no, I am wrong on the second point.  My new code would also evaluate `i = N +1`

Reply:
My bad on the ambiguous message point - you are correct, it does the same. For some reason, in that debug statement, I read it as if it would say  'x or y' in the final outcome!